builds/misc/templates/mqtt-perf-setup.yaml

```diff
@@ -0,0 +1,132 @@
+# Template to set up test host (pull and start 'mqttd' container)
+# as well as publisher node and subscriber node (mqtt-benchmark tool writen in go).
+
+jobs:
+- job: setup_host
+  pool:
+    name: $(pool.name)
+    demands: mqtt-perf-tests-pub
+  displayName: Setup Host
+  steps:
+    - checkout: none
+
+    # get latest docker image name from mqtt images pipeline, 
+    # if no image specified in the pipeline variables
+    - download: images
+      artifact: image$(arch)
+      displayName: Get MQTT broker image info
+      condition: eq(variables['brokerImageName'], '')
+
+    - script: |
+        brokerImageName=$(cat $(Pipeline.Workspace)/images/image$(arch)/image$(arch))
+        echo "##vso[task.setvariable variable=brokerImageName;]$brokerImageName"
+      displayName: 'Set broker image name variable'
+      condition: eq(variables['brokerImageName'], '')
+
+    # AMD64 setup
+    - task: SSH@0
+      displayName: Cleanup running containers AMD64
+      inputs:
+        sshEndpoint: iotedge-mqtt-perf-amd64-conn
+        runOptions: commands        
+        commands: |
+          docker kill $(docker ps -q) 2>/dev/null
+      # ignore cleanup errors if no containers running
+      continueOnError: true 
+      condition: eq(variables['arch'], 'amd64')
+
+    - task: SSH@0
+      displayName: Install and Run docker image AMD64
+      inputs:
+        sshEndpoint: iotedge-mqtt-perf-amd64-conn
+        runOptions: commands
+        commands: |
+          echo "$(registry.password)" | docker login $(registry.address) --username $(registry.user) --password-stdin 2>/dev/null
+          docker pull $(brokerImageName)
+          docker run -d -p 1883:1883 $(brokerImageName)
+          docker logout
+      condition: eq(variables['arch'], 'amd64')
+
+    # ARM32 setup
+    - task: SSH@0
+      displayName: Cleanup running containers ARM32
+      inputs:
+        sshEndpoint: iotedge-mqtt-perf-arm32v7-conn
+        runOptions: commands        
+        commands: |
+          docker kill $(docker ps -q) 2>/dev/null
```

Comment:
we normally clean up before run test.  What we do is to uninstall iotedged and remove all docker containers and do docker system prune to remove all docker images.

docker kill remove will remove all running containers.  If iotedged is still running, it will try to create those containers again.

Reply:
Good point. I'll take a look into that.

Reply:
I've added `docker system prune -a` - it should take case of all stopped containers and images.